mesh/nurbs.hpp

```diff
@@ -29,66 +29,129 @@ namespace mfem
 class GridFunction;
 
 
+/** @brief A vector of knots in one dimension, with B-spline basis functions of
+    a prescribed order. */
 class KnotVector
 {
 protected:
    static const int MaxOrder;
 
+   /// Stores the values of all knots.
    Vector knot;
-   int Order, NumOfControlPoints, NumOfElements;
+
+   /// Order or degree of the B-spline basis functions.
+   int Order;
+
+   /// Number of control points.
+   int NumOfControlPoints;
+
+   /// Number of elements, defined by distinct knots.
+   int NumOfElements;
 
 public:
-   /// Create KnotVector
+   /// Create an empty KnotVector.
    KnotVector() { }
+
+   /// Create a KnotVector by reading data from stream @a input.
    KnotVector(std::istream &input);
-   KnotVector(int Order_, int NCP);
+
+   /** @brief Create a KnotVector with undefined knots (initialized to -1) of
+       order @a order and number of control points @a NCP. */
+   KnotVector(int order, int NCP);
+
+   /// Copy constructor.
    KnotVector(const KnotVector &kv) { (*this) = kv; }
 
    KnotVector &operator=(const KnotVector &kv);
 
+   /// Return the number of elements, defined by distinct knots.
    int GetNE()    const { return NumOfElements; }
-   int GetNKS()   const { return NumOfControlPoints - Order; }
+
+   /// Return the number of control points.
    int GetNCP()   const { return NumOfControlPoints; }
+
+   /// Return the order (degree).
```

Comment:
I am personally fine with `order(degree)`, but I think this is technically incorrect? Might be wrong on this? But I think one is the other + 1. I do not know which is which, it might also be a  difference between spline/CAD vs polynomial/numerics people.


More an aside then a real comment perhaps...

Reply:
Good point. I added a reference to the NURBS book, section 2.2, which defines what we mean by order. I also removed mentions of "degree". I must have added these comments about "degree" because of functions like `DegreeElevate`. Functions like this make interchangeable use of degree and order, and they are equivalent but not equal. I think it is less confusing now, by just using order or degree in each comment.